src/fu-debug.c

```diff
@@ -107,6 +88,35 @@ fu_debug_handler_cb (const gchar *log_domain,
 	}
 }
 
+static void
+fu_debug_ignore_cb (const gchar *log_domain,
```

Comment:
I think we either need to rename this method, or preferably, just change the `g_log_set_default_handler()` if verbose mode is changed.

Reply:
Changing `g_log_set_default_handler` was actually the original way I tried.  It caused segfaults whenever `g_debug` was called.

Reply:
Were you using the correct `user_data`?

Reply:
Pretty sure.  This diff on top of the current PR segfaults in Glib.

```
diff --git a/src/fu-debug.c b/src/fu-debug.c
index 5d0e0c69..2998681d 100644
--- a/src/fu-debug.c
+++ b/src/fu-debug.c
@@ -21,6 +21,12 @@ typedef struct {
        gchar           **daemon_verbose;
 } FuDebug;
 
+static gboolean
+fu_debug_post_parse_hook (GOptionContext *context,
+                         GOptionGroup *group,
+                         gpointer data,
+                         GError **error);
+
 static void
 fu_debug_free (FuDebug *self)
 {
@@ -100,7 +106,8 @@ fu_debug_ignore_cb (const gchar *log_domain,
        if (self->verbose ||
            g_strcmp0 (g_getenv ("FWUPD_VERBOSE"), "1") == 0) {
                self->verbose = 1;
-               fu_debug_handler_cb (log_domain, log_level, message, user_data);
+//             fu_debug_handler_cb (log_domain, log_level, message, user_data);
+               fu_debug_post_parse_hook (NULL, NULL, user_data, NULL);
                return;
        }
```

```
#0  0x00007ffff7423fb7 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff7405535 in __GI_abort () at abort.c:79
#2  0x00007ffff7a32daf in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#3  0x00007ffff7a6cc5d in g_logv () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#4  0x00007ffff7a6ce2f in g_log () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#5  0x0000555555588b92 in fu_debug_post_parse_hook (context=0x0, group=0x0, data=0x5555555fb800, error=0x0) at ../src/fu-debug.c:180
#6  0x00005555555888bb in fu_debug_ignore_cb (log_domain=0x5555555a6034 "FuMain", log_level=G_LOG_LEVEL_DEBUG, message=0x5555556d0e00 "Called GetDevices()", user_data=0x5555555fb800)
    at ../src/fu-debug.c:110
#7  0x00007ffff7a6cc3f in g_logv () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#8  0x00007ffff7a6ce2f in g_log () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#9  0x00005555555840b5 in fu_main_daemon_method_call
    (connection=0x5555556b1010, sender=0x7fffe80096a0 ":1.671", object_path=0x7fffe800d7c0 "/", interface_name=0x7fffe800d640 "org.freedesktop.fwupd", method_name=0x7fffe8009b30 "GetDevices", parameters=0x7fffe800a890, invocation=0x7fffe800e470, user_data=0x5555555fb9b0) at ../src/fu-main.c:794
#10 0x00007ffff7c93ca9 in  () at /lib/x86_64-linux-gnu/libgio-2.0.so.0
#11 0x00007ffff7a65958 in g_main_context_dispatch () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#12 0x00007ffff7a65d48 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#13 0x00007ffff7a66042 in g_main_loop_run () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#14 0x000055555558713d in main (argc=1, argv=0x7fffffffe608) at ../src/fu-main.c:1637
```

Reply:
I've opted to rename